core/src/main/scala/tech/beshu/ror/accesscontrol/request/EsRequestContext.scala

```diff
@@ -99,28 +109,28 @@ class EsRequestContext private (rInfo: RequestInfoShim) extends RequestContext {
       }
       .toSet
 
-  override val templateIndicesPatterns: Set[IndexName] =
+  override lazy val templateIndicesPatterns: Set[IndexName] =
     rInfo.extractTemplateIndicesPatterns.flatMap(IndexName.fromString)
 
-  override val repositories: Set[IndexName] =
+  override lazy val repositories: Set[IndexName] =
     rInfo.extractRepositories.flatMap(IndexName.fromString)
 
-  override val snapshots: Set[IndexName] =
+  override lazy val snapshots: Set[IndexName] =
     rInfo.extractSnapshots.flatMap(IndexName.fromString)
 
-  override val isReadOnlyRequest: Boolean =
+  override lazy val isReadOnlyRequest: Boolean =
     rInfo.extractIsReadRequest
 
-  override val involvesIndices: Boolean =
+  override lazy val involvesIndices: Boolean =
     rInfo.involvesIndices
 
-  override val isCompositeRequest: Boolean =
+  override lazy val isCompositeRequest: Boolean =
     rInfo.extractIsCompositeRequest
 
-  override val isAllowedForDLS: Boolean =
+  override lazy val isAllowedForDLS: Boolean =
     rInfo.extractIsAllowedForDLS
 
-  override val hasRemoteClusters: Boolean =
+  override lazy val hasRemoteClusters: Boolean =
```

Comment:
I love all these lazy vals, it could make a nice performance boost, right?

Reply:
potentially yes, but I'd not expect big gain

Reply:
Mainly I expect it would speed up when you only use very few rules in the whole ACL